Automattic/vip-go-mu-plugins
# Move the object-cache files from stacks to mu-plugins

## Description
All `object-cache-*.php` logic should live in MU-plugins for transparency. Fixes [#696](https://github.com/Automattic/vip-go-mu-plugins/issues/696) and https://github.com/Automattic/vip-go-mu-plugins/issues/2077.

We'll need to update the current `object-cache.php` in stacks with the below contents:

```
<?php

/*
Plugin Name: Memcached
Description: Memcached backend for the WP Object Cache.
Version: 4.0.0
Plugin URI: http://wordpress.org/extend/plugins/memcached/
Author: Automattic

Install this file to wp-content/object-cache.php
*/

if ( file_exists( ABSPATH . '/wp-content/mu-plugins/drop-ins/object-cache/object-cache-stable.php' && file_exists( ABSPATH . '/wp-content/mu-plugins/drop-ins/object-cache/object-cache-next.php' ) ) ) {
	define( 'VIP_OBJECT_CACHE_DROPIN_STABLE', ABSPATH . '/wp-content/mu-plugins/drop-ins/object-cache/object-cache-stable.php' );
	define( 'VIP_OBJECT_CACHE_DROPIN_NEXT', ABSPATH . '/wp-content/mu-plugins/drop-ins/object-cache/object-cache-next.php' );
} else {
	// TODO: Remove after mu-plugins deploy
	define( 'VIP_OBJECT_CACHE_DROPIN_STABLE', __DIR__ . '/object-cache-stable.php' );
	define( 'VIP_OBJECT_CACHE_DROPIN_NEXT', __DIR__ . '/object-cache-next.php' );
}

// Load the next version on specified environment types
if ( ! defined( 'VIP_USE_NEXT_OBJECT_CACHE_DROPIN' ) ) {
	if ( in_array( VIP_GO_APP_ENVIRONMENT, [ 'develop', 'preprod', 'staging' ], true ) ) {
		define( 'VIP_USE_NEXT_OBJECT_CACHE_DROPIN', true );
	}
}

// If site is testing next version of object cache dropin, load it
if ( defined( 'VIP_USE_NEXT_OBJECT_CACHE_DROPIN' ) && true === VIP_USE_NEXT_OBJECT_CACHE_DROPIN && is_readable( VIP_OBJECT_CACHE_DROPIN_NEXT ) ) {
	require_once( VIP_OBJECT_CACHE_DROPIN_NEXT );
} else {
	// Else, load stable cache dropin (default)
	require_once( VIP_OBJECT_CACHE_DROPIN_STABLE );
}

if ( file_exists( ABSPATH . '/wp-content/mu-plugins/lib/class-apc-cache-interceptor.php' ) ) {
	require_once( ABSPATH . '/wp-content/mu-plugins/lib/class-apc-cache-interceptor.php' );
}

```
## Changelog Description

### Plugin Updated: Object Cache

Load object-cache.php from MU-plugins version.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Load up VIP Go sandbox
2) Change contents of `wp-content/object-cache.php` with the snippet from the Description
3) Check out PR so that the `object-cache` folder exists now in mu-plugins
